### DISCUSSION TITLE
Enhancements to Table Plugin

### DISCUSSION THREAD

**Discussion by stevetracvc** [apache/superset#27241] [General]
I'm finally bringing this over from the old superset-ui repo.

I have several other enhancements I've made to the table plugin (drag n drop column rearrangement will be a separate issue) that my coworkers have liked. I'm opening this up to a discussion on whether any of the features would be good additions for everyone. A few are still WIP but most seem to be working as expected. Unfortunately, while many of them don't depend on each other, they do affect each other, so it would take a little effort for me to make them all separate PRs. I don't mind doing it, but I'd need to see which things you'd want before trying to untangle them.

Here's the list:

1. A row numbers column, where the numbers stay fixed even after sorting the other columns
2. Sticky columns, like Excel Freeze Columns (WIP)
3. Two-level Column Headers, allowing you to group columns together with a higher-level header (WIP)

**Videos**

**Row Numbers**
![row_number_vid_1](https://user-images.githubusercontent.com/70416691/160241668-9e8f6ca4-d3b2-4767-837f-7e418df7d069.gif)

![row_number_vid_2](https://user-images.githubusercontent.com/70416691/160241683-ed69d5cb-df44-4067-84ac-30e969256e69.gif)

**Sticky Columns**
![freeze_cols](https://user-images.githubusercontent.com/70416691/160241704-2817c6bf-c56b-4901-89d1-4e11c3140ae5.gif)

**Column Grouping/Headers**
adding:
https://user-images.githubusercontent.com/70416691/144466015-9ee07130-493f-4fe7-b0a9-c22e000cce1c.mp4

deleting:
https://user-images.githubusercontent.com/70416691/144466126-822335f9-4273-48b9-a620-388edecd14b8.mp4

**Known issues:**

1. row numbers + two-level column header can result in weird row number location
2. sticky columns currently requires sticky column widths to be manually set
3. sticky columns doesn't work with cell bars (background is set to white, which is overridden by cell bars)

**Comment by KarenWeis** [comment]
@stevetracvc - any updates on this? I'm looking for the two-level column header update.

**Comment by stevetracvc** [comment]
@KarenWeis  No one has really expressed much interest

I haven't done anything with it since before the monorepo change, so it's still in my superset-ui fork
https://github.com/stevetracvc/superset-ui/tree/feat/table-header-group-columns

**Comment by genghistron84** [comment]
@stevetracvc - two-level column headers is a great feature and one I would definitely use for a project I'm currently working on.

**Comment by Nimit-Shah** [comment]
Hi @stevetracvc Can you release the ui change you did for freeze columns? or provide a branch link?

**Comment by ambarishp** [comment]
Hi @stevetracvc  i m looking for column grouping and freeze column features. can this be released?
